charts/k8s-monitoring/test/config-analysis.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -eo pipefail
+
+AGENT_HOST="${AGENT_HOST:-http://localhost:8080}"
+sources=$1
+
+function discoveryRelabel() {
+    component=$1
```

Comment:
May not be necessary but I think there might be some scope leakage here, so you could use `local`?

```suggestion
    local component=$1
```

Reply:
Good call.